cmd/build.go

```diff
@@ -0,0 +1,203 @@
+package cmd
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/CircleCI-Public/circleci-cli/settings"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var dockerAPIVersion = "1.23"
+
+var job = "build"
+var nodeTotal = 1
+var checkoutKey = "~/.ssh/id_rsa"
+var skipCheckout = true
+
+func newBuildCommand() *cobra.Command {
+
+	command := &cobra.Command{
+		Use:   "build",
+		Short: "Run a build",
+		RunE:  runBuild,
+	}
+
+	/*
+		TODO: Support these flags:
+				--branch string         Git branch
+			-e, --env -e VAR=VAL        Set environment variables, e.g. -e VAR=VAL
+				--index int             node index of parallelism
+				--repo-url string       Git Url
+				--revision string       Git Revision
+			-v, --volume stringSlice    Volume bind-mounting
+	*/
+
+	flags := command.Flags()
+	flags.StringVarP(&dockerAPIVersion, "docker-api-version", "d", dockerAPIVersion, "The Docker API version to use")
```

Comment:
I don't think you want to duplicate all the flags here. There is a better way!

One option is to use `flags.SetIntersperced(false)` https://godoc.org/github.com/spf13/pflag#SetInterspersed. That way any flags after a positional argument (or a `--`) are kept in `Args`. It might also work for any not recognized flags, I don't remember.

You're already appending args so that should be it.

Reply:
Oh sweet, that will make things easier 👌